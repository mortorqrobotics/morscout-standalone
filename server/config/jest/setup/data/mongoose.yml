Season:
  - _id: 5d7be5a7994e35048b9e5e0e
    year: 2019
    name: Destination Deep Space
    buildStart: 2019-01-04 12:00:00 -5
    buildEnd: 2019-01-06 12:00:00 -5
    seasonStart: 2019-01-08 12:00:00 -5
    seasonEnd: 2019-02-03 12:00:00 -5
  - _id: 4d7be5a7994e35048b9e5e0e
    year: 2015
    name: Destination Deep Space
    buildStart: 2019-01-04 12:00:00 -5
    buildEnd: 2019-01-06 12:00:00 -5
    seasonStart: 2019-01-08 12:00:00 -5
    seasonEnd: 2019-02-03 12:00:00 -5
Regional:
  - _id: 6d7be5a7994e35048b9e5e0e
    name: Hi!!!
    key: abc
    url: google.com
    eventType: regional
    startDate: 2019-01-08 12:00:00 -5
    endDate: 2019-02-03 12:00:00 -5
    season: 5d7be5a7994e35048b9e5e0e
    location:
      type: Point
      coordinates: [0,0]
Alliance:
  - _id: 2d7ce5a7994e35048b9e5e0e
    teams:
      - 1d7ce5a7994e35048b9e5e0e
      - 1d7ce5a7994e35048b9e5e0e
      - 1d7ce5a7994e35048b9e5e0e
Match:
  - _id: 7d7be5a7994e35048b9e5e0e
    redAlliance: 2d7ce5a7994e35048b9e5e0e
    blueAlliance: 2d7ce5a7994e35048b9e5e0e
    winner: Red
    startDate: 2019-02-03 12:00:00 -5
    endDate: 2019-02-03 12:10:00 -5
    Regional: 6d7be5a7994e35048b9e5e0e
User:
  - _id: 8d7be5a7994e35048b9e5e0f
    team: 9d7be5a7994e35048b9e5e0e
    phone: 1334567890
    email: bo@bob.bob
    firstname: Bob
    lastname: Boby
    password: Secret
    username: bob2
Team:
  - _id: 9d7be5a7994e35048b9e5e0e
    regional: 6d7be5a7994e35048b9e5e0e
    number: 1515
    name: MorTorq
    location:
      type: Point
      coordinates: [0,0]
