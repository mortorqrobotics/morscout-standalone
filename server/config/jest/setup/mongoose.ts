import mongoose from "mongoose";
import { MongoMemoryServer } from "mongodb-memory-server";
import getData from "./getData";

let mongoServer: MongoMemoryServer;
export async function setup() {
  mongoServer = new MongoMemoryServer();
  const mongoUri = await mongoServer.getConnectionString();
  (process as any).mongoUri = mongoUri;
  await mongoose.connect(
    mongoUri,
    { useNewUrlParser: true, useCreateIndex: true },
    err => {
      if (err) console.error(err);
    }
  );
  await getData();
  return mongoUri;
}

export async function teardown() {
  mongoServer.stop();
}
