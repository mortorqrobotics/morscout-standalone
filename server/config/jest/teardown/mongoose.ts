import mongoose from "mongoose";

module.exports = () => {
  mongoose.disconnect();
  (mongoose as any).server.stop();
};
