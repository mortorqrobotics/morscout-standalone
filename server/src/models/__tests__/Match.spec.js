import "mockingoose";
import mongoose from "mongoose";
import schema from "../Match";

test("Model is same throughout", async () => {
  expect.assertions(2);
  expect(schema).toMatchSnapshot();
  expect(mongoose.model("Match", schema)).toMatchSnapshot();
});
