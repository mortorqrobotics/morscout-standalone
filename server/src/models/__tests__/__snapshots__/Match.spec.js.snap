// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model is same throughout 1`] = `
Schema {
  "$id": 1,
  "_indexes": Array [],
  "_userProvidedOptions": Object {},
  "aliases": Object {},
  "callQueue": Array [],
  "childSchemas": Array [],
  "inherits": Object {},
  "methodOptions": Object {},
  "methods": Object {},
  "nested": Object {},
  "obj": Object {
    "Regional": Object {
      "ref": "Regional",
      "type": [Function],
    },
    "blueAlliance": Object {
      "ref": "Alliance",
      "type": [Function],
    },
    "endData": Object {
      "type": [Function],
      "validate": [Function],
    },
    "redAlliance": Object {
      "ref": "Alliance",
      "type": [Function],
    },
    "startDate": Object {
      "type": [Function],
    },
    "winner": Object {
      "default": null,
      "enum": Array [
        "Red",
        "Blue",
      ],
      "type": [Function],
    },
  },
  "options": Object {
    "_id": true,
    "autoIndex": null,
    "bufferCommands": true,
    "capped": false,
    "discriminatorKey": "__t",
    "id": true,
    "minimize": true,
    "noId": false,
    "noVirtualId": false,
    "read": null,
    "shardKey": null,
    "strict": true,
    "typeKey": "type",
    "validateBeforeSave": true,
    "versionKey": "__v",
  },
  "paths": Object {
    "Regional": ObjectId {
      "_index": null,
      "getters": Array [],
      "instance": "ObjectID",
      "options": Object {
        "ref": "Regional",
        "type": [Function],
      },
      "path": "Regional",
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "_id": ObjectId {
      "_index": null,
      "defaultValue": [Function],
      "getters": Array [],
      "instance": "ObjectID",
      "options": Object {
        "auto": true,
        "type": [Function],
      },
      "path": "_id",
      "setters": Array [
        [Function],
      ],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "blueAlliance": ObjectId {
      "_index": null,
      "getters": Array [],
      "instance": "ObjectID",
      "options": Object {
        "ref": "Alliance",
        "type": [Function],
      },
      "path": "blueAlliance",
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "endData": SchemaDate {
      "_index": null,
      "getters": Array [],
      "instance": "Date",
      "options": Object {
        "type": [Function],
        "validate": [Function],
      },
      "path": "endData",
      "setters": Array [],
      "validators": Array [
        Object {
          "message": "Validator failed for path \`{PATH}\` with value \`{VALUE}\`",
          "type": "user defined",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
    "redAlliance": ObjectId {
      "_index": null,
      "getters": Array [],
      "instance": "ObjectID",
      "options": Object {
        "ref": "Alliance",
        "type": [Function],
      },
      "path": "redAlliance",
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "startDate": SchemaDate {
      "_index": null,
      "getters": Array [],
      "instance": "Date",
      "options": Object {
        "type": [Function],
      },
      "path": "startDate",
      "setters": Array [],
      "validators": Array [],
      Symbol(mongoose#schemaType): true,
    },
    "winner": SchemaString {
      "_index": null,
      "defaultValue": null,
      "enumValidator": [Function],
      "enumValues": Array [
        "Red",
        "Blue",
      ],
      "getters": Array [],
      "instance": "String",
      "options": Object {
        "default": null,
        "enum": Array [
          "Red",
          "Blue",
        ],
        "type": [Function],
      },
      "path": "winner",
      "regExp": null,
      "setters": Array [],
      "validators": Array [
        Object {
          "enumValues": Array [
            "Red",
            "Blue",
          ],
          "message": "\`{VALUE}\` is not a valid enum value for path \`{PATH}\`.",
          "type": "enum",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
  },
  "plugins": Array [],
  "query": Object {},
  "s": Object {
    "hooks": Kareem {
      "_posts": Map {},
      "_pres": Map {},
    },
  },
  "singleNestedPaths": Object {},
  "statics": Object {},
  "subpaths": Object {},
  "tree": Object {
    "Regional": Object {
      "ref": "Regional",
      "type": [Function],
    },
    "_id": Object {
      "auto": true,
      "type": [Function],
    },
    "blueAlliance": Object {
      "ref": "Alliance",
      "type": [Function],
    },
    "endData": Object {
      "type": [Function],
      "validate": [Function],
    },
    "redAlliance": Object {
      "ref": "Alliance",
      "type": [Function],
    },
    "startDate": Object {
      "type": [Function],
    },
    "winner": Object {
      "default": null,
      "enum": Array [
        "Red",
        "Blue",
      ],
      "type": [Function],
    },
  },
  "virtuals": Object {},
}
`;

exports[`Model is same throughout 2`] = `[Function]`;
